tests/connection.rs

```diff
@@ -220,28 +209,8 @@ fn deferred_replication() {
 
     server_app.connect_client(&mut client_app);
 
-    let connected_clients = server_app.world().resource::<ConnectedClients>();
-    assert!(!connected_clients.is_empty());
-
-    let replicated_clients = server_app.world().resource::<ReplicatedClients>();
-    assert!(
-        replicated_clients.is_empty(),
-        "server shouldn't replicate to yet"
-    );
-
-    let client = client_app.world().resource::<RepliconClient>();
-    let client_id = client.id().unwrap();
-    server_app.world_mut().trigger(StartReplication(client_id));
-
-    let replicated_clients = server_app.world().resource::<ReplicatedClients>();
-    assert!(
-        !replicated_clients.is_empty(),
-        "server now should start replicating"
-    );
-
-    // Make sure that enabling replication twice do nothing.
-    server_app.world_mut().trigger(StartReplication(client_id));
-
-    let replicated_clients = server_app.world().resource::<ReplicatedClients>();
-    assert_eq!(replicated_clients.len(), 1);
+    server_app
```

Comment:
Since it's no longer a trigger, I just check if `ReplicatedClient` is not inserted if `replicate_after_connect` is inserted.